index.d.ts

```diff
@@ -2,33 +2,29 @@ export { } from "./jsx";
 
 export as namespace skate;
 
-export interface HasProps<El extends Component> {
-  props: { [key: string]: skate.PropOptions<El, any>; };
-}
-
-export interface OnUpdatedCallback {
-  updatedCallback(previousProps: any): boolean | undefined | void;
-}
-
-export interface OnRenderCallback {
-  renderCallback(): any;
-}
-
-export interface OnRenderedCallback {
-  renderedCallback(): void;
-}
-
-export class Component extends HTMLElement {
-  // static readonly props: { [name: string]: PropAttr<Component, any> };
+export class Component extends HTMLElement implements ICustomElementV1, ISkateLivecycle {
+  static readonly props: { [nameOrSymbol: string]: PropOptions<El, any> };
```

Comment:
I want to add this line too.
But this definition can't compiling [this code](https://github.com/vvakame/til/blob/49294d3185df832f93c01f5b34f811671d99a6ea/skatejs-newdef/src/prop-examples.tsx#L7-L22).